web-api/src/storage/postgres.rs

```diff
@@ -353,38 +351,33 @@ impl Database {
         Ok(())
     }
 
-    async fn upsert_interactions<'d, I>(
+    async fn upsert_interactions(
         tx: &mut Transaction<'_, Postgres>,
         user_id: &UserId,
         now: DateTime<Utc>,
-        interactions: I,
-    ) -> Result<(), Error>
-    where
-        I: IntoIterator<Item = (&'d DocumentId, UserInteractionType)>,
-        I::IntoIter: ExactSizeIterator,
-    {
-        let interactions = interactions.into_iter();
-        if interactions.len() == 0 {
-            return Ok(());
-        }
+        interactions: &HashMap<&DocumentId, (&InteractedDocument, UserInteractionType)>,
+    ) -> Result<(), Error> {
```

Comment:
As this was created from a map before this couldn't trigger the bug before, I still changed it to avoid running into the bug again in the future.